contracts/LotteryContractUpdated.sol

```diff
@@ -0,0 +1,135 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+contract Lottery {
+    
+    enum UserType {Issuer,Investor}
+    enum LotteryState {NotStarted, Started,Finished}
+    
+    LotteryState public state;
+    
+    struct Player {
+        address payable playerAddress;
+        string playerName;
+        string emailId;
+        string phoneNumber;
+        bool isExist;
+    }
+    
+    struct LotteryTicket {
+        uint lotteryId;
+        uint price;
+        address payable ticketOwner;
+    }
+    
+
+    address private owner;
+    uint32 private totalAmount;
+    Player[] public winner;
+    
+    mapping (address => Player) playerMap;
+    Player[] players;
+    LotteryTicket[] tickets;
+
+    constructor () public {
+        state = LotteryState.NotStarted;
+        owner = msg.sender;
+    }
+
+    function registerPlayer(string memory _name, string memory _emailId, string memory  _phoneNumber) public payable returns (string memory result) {
+        //Owner shouldn't be player
+        require(owner != msg.sender);
+        require((state != LotteryState.NotStarted) && (state != LotteryState.Finished), "Player can't register while lottery has started or finished.");
+        
+        result = "FAIL !!!";
+        if(!playerMap[msg.sender].isExist ) {
+            Player memory _player = Player(msg.sender, _name, _emailId, _phoneNumber,true);
+            playerMap[msg.sender] = _player;
+            players.push(_player);
+            result = "Registered Successfully !!!";
+        }
+
+        return(result);
+        // new event to be send to owner
+    }
+    
+    function startLottery() public{
+        require (owner == msg.sender, "Only owner can open lottery process.");
+        state = LotteryState.Started;
+        
+    }
+
+    function stopLottery() public{
+        require (owner == msg.sender, "Only owner can open lottery process.");
+        state = LotteryState.Finished;
+        
+    }
+    
+    function buyLotteryTickets() public payable returns (string memory result) {
```

Comment:
Better add a parameter for accepting the no of tickets that the message sender intends to buy so that we can add support for the same in the screens too.

Reply:
Will check